### DISCUSSION TITLE
BTicino/Legrand devices rejoining continuosly

### DISCUSSION THREAD

**Discussion by xraver** [Koenkk/zigbee2mqtt#20755] [Devices]
Hi,
in my system (intel nuc with proxmox + docker and Sonoff dongle E) I have about 10 devices. Most of them are xiaomi sensors (doors + temperature), 2 sonoff mini extreme and 2 bticino (smart dimmer + covers),
I've bought two new devices: Legrand 064888/BTicino 3584C and light switch (BTicino NT4003C).
I've tried to join the two new devices and both of them have the same behaviour: they join the network, the system recognize the device and after a couple of seconds they left the network.
I've replaced one of them to understand if the issue was the device and with the new one same issue.

While the devices are available in the system I can succesfully control them (e.g. turn on light).

my dongle is Zigbee 3.0 USB Dongle Plus ("ZBDongle-E") with firmware 6.7.9.0 build 405.
can you help me? thanks

attached an example of log:
`info  2024-01-12 16:35:08: Zigbee: allowing new devices to join.
info  2024-01-12 16:35:08: MQTT publish: topic 'zigbee2mqtt/bridge/response/permit_join', payload '{"data":{"time":254,"value":true},"status":"ok","transaction":"9arn6-1"}'
info  2024-01-12 16:35:09: Device '0x000474000025912a' joined
info  2024-01-12 16:35:09: MQTT publish: topic 'zigbee2mqtt/bridge/event', payload '{"data":{"friendly_name":"0x000474000025912a","ieee_address":"0x000474000025912a"},"type":"device_joined"}'
info  2024-01-12 16:35:09: Starting interview of '0x000474000025912a'
info  2024-01-12 16:35:09: MQTT publish: topic 'zigbee2mqtt/bridge/event', payload '{"data":{"friendly_name":"0x000474000025912a","ieee_address":"0x000474000025912a","status":"started"},"type":"device_interview"}'
info  2024-01-12 16:35:10: Successfully interviewed '0x000474000025912a', device has successfully been paired
info  2024-01-12 16:35:10: Device '0x000474000025912a' is supported, identified as: Legrand Wired micromodule switch (064888)
info  2024-01-12 16:35:10: MQTT publish: topic 'zigbee2mqtt/bridge/event', payload '{"data":{"definition":{"description":"Wired micromodule switch","exposes":[{"features":[{"access":7,"description":"On/off state of the switch","label":"State","name":"state","property":"state","type":"binary","value_off":"OFF","value_on":"ON","value_toggle":"TOGGLE"}],"type":"switch"},{"access":7,"description":"Controls the behavior when the device is powered on after power loss. If you get an `UNSUPPORTED_ATTRIBUTE` error, the device does not support it.","label":"Power-on behavior","name":"power_on_behavior","property":"power_on_behavior","type":"enum","values":["off","on","toggle","previous"]},{"access":1,"description":"Link quality (signal strength)","label":"Linkquality","name":"linkquality","property":"linkquality","type":"numeric","unit":"lqi","value_max":255,"value_min":0}],"model":"064888","options":[{"access":2,"description":"State actions will also be published as 'action' when true (default false).","label":"State action","name":"state_action","property":"state_action","type":"binary","value_off":false,"value_on":true}],"supports_ota":true,"vendor":"Legrand"},"friendly_name":"0x000474000025912a","ieee_address":"0x000474000025912a","status":"successful","supported":true},"type":"device_interview"}'
info  2024-01-12 16:35:10: Configuring '0x000474000025912a'
info  2024-01-12 16:35:10: MQTT publish: topic 'homeassistant/switch/0x000474000025912a/switch/config', payload '{"availability":[{"topic":"zigbee2mqtt/bridge/state","value_template":"{{ value_json.state }}"}],"command_topic":"zigbee2mqtt/0x000474000025912a/set","device":{"identifiers":["zigbee2mqtt_0x000474000025912a"],"manufacturer":"Legrand","model":"Wired micromodule switch (064888)","name":"0x000474000025912a","sw_version":"002c","via_device":"zigbee2mqtt_bridge_0xe0798dfffe8858d8"},"name":null,"object_id":"0x000474000025912a","origin":{"name":"Zigbee2MQTT","sw":"1.35.1","url":"https://www.zigbee2mqtt.io"},"payload_off":"OFF","payload_on":"ON","state_topic":"zigbee2mqtt/0x000474000025912a","unique_id":"0x000474000025912a_switch_zigbee2mqtt","value_template":"{{ value_json.state }}"}'
info  2024-01-12 16:35:10: MQTT publish: topic 'homeassistant/sensor/0x000474000025912a/power_on_behavior/config', payload '{"availability":[{"topic":"zigbee2mqtt/bridge/state","value_template":"{{ value_json.state }}"}],"device":{"identifiers":["zigbee2mqtt_0x000474000025912a"],"manufacturer":"Legrand","model":"Wired micromodule switch (064888)","name":"0x000474000025912a","sw_version":"002c","via_device":"zigbee2mqtt_bridge_0xe0798dfffe8858d8"},"enabled_by_default":false,"entity_category":"diagnostic","icon":"mdi:power-settings","name":"Power-on behavior","object_id":"0x000474000025912a_power_on_behavior","origin":{"name":"Zigbee2MQTT","sw":"1.35.1","url":"https://www.zigbee2mqtt.io"},"state_topic":"zigbee2mqtt/0x000474000025912a","unique_id":"0x000474000025912a_power_on_behavior_zigbee2mqtt","value_template":"{{ value_json.power_on_behavior }}"}'
info  2024-01-12 16:35:10: MQTT publish: topic 'homeassistant/select/0x000474000025912a/power_on_behavior/config', payload '{"availability":[{"topic":"zigbee2mqtt/bridge/state","value_template":"{{ value_json.state }}"}],"command_topic":"zigbee2mqtt/0x000474000025912a/set/power_on_behavior","device":{"identifiers":["zigbee2mqtt_0x000474000025912a"],"manufacturer":"Legrand","model":"Wired micromodule switch (064888)","name":"0x000474000025912a","sw_version":"002c","via_device":"zigbee2mqtt_bridge_0xe0798dfffe8858d8"},"entity_category":"config","icon":"mdi:power-settings","name":"Power-on behavior","object_id":"0x000474000025912a_power_on_behavior","options":["off","on","toggle","previous"],"origin":{"name":"Zigbee2MQTT","sw":"1.35.1","url":"https://www.zigbee2mqtt.io"},"state_topic":"zigbee2mqtt/0x000474000025912a","unique_id":"0x000474000025912a_power_on_behavior_zigbee2mqtt","value_template":"{{ value_json.power_on_behavior }}"}'
info  2024-01-12 16:35:10: MQTT publish: topic 'homeassistant/sensor/0x000474000025912a/linkquality/config', payload '{"availability":[{"topic":"zigbee2mqtt/bridge/state","value_template":"{{ value_json.state }}"}],"device":{"identifiers":["zigbee2mqtt_0x000474000025912a"],"manufacturer":"Legrand","model":"Wired micromodule switch (064888)","name":"0x000474000025912a","sw_version":"002c","via_device":"zigbee2mqtt_bridge_0xe0798dfffe8858d8"},"enabled_by_default":false,"entity_category":"diagnostic","icon":"mdi:signal","name":"Linkquality","object_id":"0x000474000025912a_linkquality","origin":{"name":"Zigbee2MQTT","sw":"1.35.1","url":"https://www.zigbee2mqtt.io"},"state_class":"measurement","state_topic":"zigbee2mqtt/0x000474000025912a","unique_id":"0x000474000025912a_linkquality_zigbee2mqtt","unit_of_measurement":"lqi","value_template":"{{ value_json.linkquality }}"}'
info  2024-01-12 16:35:10: MQTT publish: topic 'homeassistant/sensor/0x000474000025912a/update_state/config', payload '{"availability":[{"topic":"zigbee2mqtt/bridge/state","value_template":"{{ value_json.state }}"}],"device":{"identifiers":["zigbee2mqtt_0x000474000025912a"],"manufacturer":"Legrand","model":"Wired micromodule switch (064888)","name":"0x000474000025912a","sw_version":"002c","via_device":"zigbee2mqtt_bridge_0xe0798dfffe8858d8"},"enabled_by_default":false,"entity_category":"diagnostic","icon":"mdi:update","name":"Update state","object_id":"0x000474000025912a_update_state","origin":{"name":"Zigbee2MQTT","sw":"1.35.1","url":"https://www.zigbee2mqtt.io"},"state_topic":"zigbee2mqtt/0x000474000025912a","unique_id":"0x000474000025912a_update_state_zigbee2mqtt","value_template":"{{ value_json['update']['state'] }}"}'
info  2024-01-12 16:35:10: MQTT publish: topic 'homeassistant/binary_sensor/0x000474000025912a/update_available/config', payload '{"availability":[{"topic":"zigbee2mqtt/bridge/state","value_template":"{{ value_json.state }}"}],"device":{"identifiers":["zigbee2mqtt_0x000474000025912a"],"manufacturer":"Legrand","model":"Wired micromodule switch (064888)","name":"0x000474000025912a","sw_version":"002c","via_device":"zigbee2mqtt_bridge_0xe0798dfffe8858d8"},"device_class":"update","enabled_by_default":false,"entity_category":"diagnostic","name":null,"object_id":"0x000474000025912a_update_available","origin":{"name":"Zigbee2MQTT","sw":"1.35.1","url":"https://www.zigbee2mqtt.io"},"payload_off":false,"payload_on":true,"state_topic":"zigbee2mqtt/0x000474000025912a","unique_id":"0x000474000025912a_update_available_zigbee2mqtt","value_template":"{{ value_json['update']['state'] == \"available\" }}"}'
info  2024-01-12 16:35:10: MQTT publish: topic 'homeassistant/update/0x000474000025912a/update/config', payload '{"availability":[{"topic":"zigbee2mqtt/bridge/state","value_template":"{{ value_json.state }}"}],"command_topic":"zigbee2mqtt/bridge/request/device/ota_update/update","device":{"identifiers":["zigbee2mqtt_0x000474000025912a"],"manufacturer":"Legrand","model":"Wired micromodule switch (064888)","name":"0x000474000025912a","sw_version":"002c","via_device":"zigbee2mqtt_bridge_0xe0798dfffe8858d8"},"device_class":"firmware","entity_category":"config","entity_picture":"https://github.com/Koenkk/zigbee2mqtt/raw/master/images/logo.png","json_attributes_template":"{\"in_progress\": {{ iif(value_json['update']['state'] == 'updating', 'true', 'false') }} }","json_attributes_topic":"zigbee2mqtt/0x000474000025912a","latest_version_template":"{{ value_json['update']['latest_version'] }}","latest_version_topic":"zigbee2mqtt/0x000474000025912a","name":null,"object_id":"0x000474000025912a","origin":{"name":"Zigbee2MQTT","sw":"1.35.1","url":"https://www.zigbee2mqtt.io"},"payload_install":"{\"id\": \"0x000474000025912a\"}","state_topic":"zigbee2mqtt/0x000474000025912a","unique_id":"0x000474000025912a_update_zigbee2mqtt","value_template":"{{ value_json['update']['installed_version'] }}"}'
info  2024-01-12 16:35:10: Successfully configured '0x000474000025912a'
warn  2024-01-12 16:35:16: Device '0x000474000025912a' left the network
info  2024-01-12 16:35:16: MQTT publish: topic 'zigbee2mqtt/bridge/event', payload '{"data":{"friendly_name":"0x000474000025912a","ieee_address":"0x000474000025912a"},"type":"device_leave"}'
warn  2024-01-12 16:35:17: Device '0x000474000025912a' left the network
info  2024-01-12 16:35:17: MQTT publish: topic 'zigbee2mqtt/bridge/event', payload '{"data":{"friendly_name":"0x000474000025912a","ieee_address":"0x000474000025912a"},"type":"device_leave"}'
info  2024-01-12 16:35:18: Device '0x000474000025912a' joined
info  2024-01-12 16:35:18: MQTT publish: topic 'zigbee2mqtt/bridge/event', payload '{"data":{"friendly_name":"0x000474000025912a","ieee_address":"0x000474000025912a"},"type":"device_joined"}'
info  2024-01-12 16:35:18: Starting interview of '0x000474000025912a'
info  2024-01-12 16:35:18: MQTT publish: topic 'zigbee2mqtt/bridge/event', payload '{"data":{"friendly_name":"0x000474000025912a","ieee_address":"0x000474000025912a","status":"started"},"type":"device_interview"}'
info  2024-01-12 16:35:19: Successfully interviewed '0x000474000025912a', device has successfully been paired
info  2024-01-12 16:35:19: Device '0x000474000025912a' is supported, identified as: Legrand Wired micromodule switch (064888)
info  2024-01-12 16:35:19: MQTT publish: topic 'zigbee2mqtt/bridge/event', payload '{"data":{"definition":{"description":"Wired micromodule switch","exposes":[{"features":[{"access":7,"description":"On/off state of the switch","label":"State","name":"state","property":"state","type":"binary","value_off":"OFF","value_on":"ON","value_toggle":"TOGGLE"}],"type":"switch"},{"access":7,"description":"Controls the behavior when the device is powered on after power loss. If you get an `UNSUPPORTED_ATTRIBUTE` error, the device does not support it.","label":"Power-on behavior","name":"power_on_behavior","property":"power_on_behavior","type":"enum","values":["off","on","toggle","previous"]},{"access":1,"description":"Link quality (signal strength)","label":"Linkquality","name":"linkquality","property":"linkquality","type":"numeric","unit":"lqi","value_max":255,"value_min":0}],"model":"064888","options":[{"access":2,"description":"State actions will also be published as 'action' when true (default false).","label":"State action","name":"state_action","property":"state_action","type":"binary","value_off":false,"value_on":true}],"supports_ota":true,"vendor":"Legrand"},"friendly_name":"0x000474000025912a","ieee_address":"0x000474000025912a","status":"successful","supported":true},"type":"device_interview"}'
info  2024-01-12 16:35:19: Configuring '0x000474000025912a'
info  2024-01-12 16:35:20: Successfully configured '0x000474000025912a'
warn  2024-01-12 16:35:27: Device '0x000474000025912a' left the network
`

**Comment by iceland12** [comment]
I'm experiencing the same issue with Legrand 412171 with Zibee2Mqtt 1.38.1 with OH4. Some of same device are correctly joined but some other not. My idea is probably their fw version are different but obviously I cannot upgrade the ones not joined.
Any idea is appreciated.

**Reply by santiagoln1998** [reply]
I've heard that when you just joined them, you need to toggle on/off on the fiscal device. I'm getting one next week so I will tell then.

**Reply by iceland12** [reply]
Thanks @santiagoln1998 ! Luckily in last two years a lot of things changed, e.g. moved from OH4 to HA.
Now I don't have this issue anymore.
